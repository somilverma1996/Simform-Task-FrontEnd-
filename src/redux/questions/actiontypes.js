import {
    ADD_QUESTION,
    GET_QUESTIONS,
  } from "./actions";

  export const addQuestion = (data) => {
    console.log(data)
    return {
      type: ADD_QUESTION,
      payload: data
    };
  };
  
  export const getQuestion = () => {
    return {
      type: GET_QUESTIONS,
    };
  };

  export const addQues = (data) => {
    return (dispatch) => {
      dispatch(addQuestion(data));
    };
  };