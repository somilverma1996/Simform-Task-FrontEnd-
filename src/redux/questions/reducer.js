import {
    ADD_QUESTION,
    GET_QUESTIONS,
  } from "./actions";

const initialState = [];

const reducer = (state = initialState, action) => {
    switch (action.type) {
      case ADD_QUESTION:
        return [
          ...action.payload
        ];
      case GET_QUESTIONS:
        return state;
      default:
        return state;
    }
  };

  export default reducer;