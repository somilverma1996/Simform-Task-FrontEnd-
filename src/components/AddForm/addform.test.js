// import 'jsdom-global/register';
import React from 'react';
import { ADD_QUESTION } from "../../redux/questions/actions";
import QuestionForm from "./index";
import { configure, shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import axios from 'axios';
import { Provider } from "react-redux";
import { cleanup, waitFor, render } from '@testing-library/react';
import store from "../../redux/store/index";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Questions', () => {
    it('render questions ', () => {
        shallow(
            <Provider store={store}>
                <QuestionForm />
            </Provider>
        )
    });
    it("renders Account header", () => {
        const wrapper = shallow(
            <Provider store={store}>
                <QuestionForm />
            </Provider>
        );
        const welcome = <div style={{ width: '70%' }} class="card text-center d-flex m-auto mt-5"></div>;
        expect(wrapper.contains(welcome)).toEqual(false);
    });
    it('render Uis ', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <QuestionForm>
                    <h2>Create a New Form</h2>
                </QuestionForm>
            </Provider>
        );
        expect(wrapper.contains(<h2>Create a New Form</h2>)).toBe(true)
    });
    describe("Matches the component snapshot made ", () => {
        it("must call the mock method with button click", () => {
            const wrapper = shallow(
                <Provider store={store}>
                    <QuestionForm id="btnSubmit" />
                </Provider>
            );
            let w = wrapper.find('#btnSubmit').hostNodes()
            expect(toJson(w)).toMatchSnapshot();

        });
    });




    it('simulates click events', () => {
        const button = shallow(
            <Provider store={store}>
                <QuestionForm>
                    <a id="btnSubmit">Submit</a>
                </QuestionForm>
            </Provider>
        );
        button.find('#btnSubmit').simulate('click');
    });

    describe('when clicking the `submit` button', () => {
        let data = {
            questions_id: "60d0a798482b46ad87012155",
            questions: [{
                answers: "2, 1, 3",
                question: "Ques 1"
            }]
        }
        const mockSubmit = jest.fn();

        const props = {
            submitData: mockSubmit
        };
        const wrapper = shallow(
            <Provider store={store}>
                <QuestionForm>
                    <a id="btnSubmit" {...props}></a>
                </QuestionForm>
            </Provider>
        );

        beforeEach(() => {
            wrapper.find('#btnSubmit').simulate('click');
        });

        it('renders the `submit` button', () => {
            expect(wrapper.find('#btnSubmit').length).toBe(1);
        });

    });


})


