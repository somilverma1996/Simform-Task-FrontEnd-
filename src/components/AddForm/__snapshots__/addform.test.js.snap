// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions Executes a handler function must call the mock method with button click 1`] = `null`;

exports[`Questions MAtches the component snapshot made  must call the mock method with button click 1`] = `null`;

exports[`Questions Matches the component snapshot made  must call the mock method with button click 1`] = `null`;

exports[`Questions Simulate click event must call the mock method with button click 1`] = `
<Connect(QuestionForm)
  id="btnSubmit"
/>
`;
